### PR TITLE
Search and replace testing for example scripts

### LINKED ISSUE #415: We should use CI to test that examples and verification tests run without errors

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Thanks @jm-c and @christophernhill for the idea.

Currently fails as examples have gone stale so I'll fix this.

I'll also add one for `internal_wave.jl`.

Resolves #415

### CODE REVIEW

**Comment by glwagner** [comment]
Should also change the size of grid I think.

**Comment by ali-ramadhan** [comment]
I think this is ready to be merged. It showcases that testing examples by rewriting them works pretty well, is easy, and should scale to work with other examples.

I actually can't test `internal_wave.jl` without rewriting it as it depends on `PyPlot` while `deepening_mixed_layer.jl` lets you set `makeplot = false`. Might be good to leave for another PR.

**Comment by ali-ramadhan** [comment]
Well, this is an easy way to increase code coverage.

**Comment by ali-ramadhan** [comment]
@suyashbire1 Can you review this PR and approve/merge if it looks okay?

Hoping to merge so we can use the testing framework with other examples.

File: test/test_examples.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+EXAMPLES_DIR = "../examples/"
+
+function run_deepening_mixed_layer_example(arch)
+    example_filepath = joinpath(EXAMPLES_DIR, "deepening_mixed_layer.jl")
+    txt = read(example_filepath, String)
+
+    arch == GPU() && (txt = replace(txt, "arch = CPU()" => "arch = GPU()"))
+
+    txt = replace(txt, "N = 32" => "N = 16")
```
Comment (glwagner) [inline test/test_examples.jl]: Fine for now, but we should write a function that auto-changes example parameters so we don't have to copy-paste these lines for each example.

**Pull request merged**